src/extensions/yfm/YfmTable/plugins/YfmTableControls/dnd/dnd-ghost.ts

```diff
@@ -0,0 +1,226 @@
+import type {EditorView} from '#pm/view';
+import type {TableDescBinded} from 'src/table-utils/table-desc';
+
+type Event = Pick<MouseEvent, 'clientX' | 'clientY' | 'target'>;
+
+type BuildGhostResult = {
+    domElement: HTMLElement;
+    shiftX: number;
+    shiftY: number;
+};
+
+export type YfmTableDnDGhostParams = {
+    initial: Event;
+    type: 'row' | 'column';
+    rangeIdx: number;
+    tableDesc: TableDescBinded;
+};
+
+export class YfmTableDnDGhost {
+    private _x: number;
+    private _y: number;
+
+    private readonly _dndBackgroundElem: HTMLElement;
+    private readonly _ghostTable: HTMLElement;
+    private readonly _ghostButton: HTMLElement | null = null;
+
+    private readonly _tblShiftX: number;
+    private readonly _tblShiftY: number;
+
+    private readonly _btnShiftX: number = 0;
+    private readonly _btnShiftY: number = 0;
+
+    private _rafId: number;
+
+    constructor(view: EditorView, params: YfmTableDnDGhostParams) {
+        this._x = params.initial.clientX;
+        this._y = params.initial.clientY;
+
+        this._dndBackgroundElem = view.dom.ownerDocument.createElement('div');
+        this._dndBackgroundElem.classList.add('g-md-yfm-table-dnd-cursor-background');
+
+        {
+            const res = this._buildGhostButton(params);
+            if (res) {
+                this._ghostButton = res.domElement;
+                this._btnShiftX = res.shiftX;
+                this._btnShiftY = res.shiftY;
+                this._dndBackgroundElem.appendChild(this._ghostButton);
+            }
+        }
+
+        {
+            const {domElement, shiftX, shiftY} =
+                params.type === 'row'
+                    ? this._buildRowGhost(view, params)
+                    : this._buildColumnGhost(view, params);
+
+            this._ghostTable = domElement;
+            this._tblShiftX = shiftX;
+            this._tblShiftY = shiftY;
+            this._dndBackgroundElem.appendChild(this._ghostTable);
+        }
+
+        this._updatePositions();
+
+        this._rafId = requestAnimationFrame(() => {
+            document.body.append(this._dndBackgroundElem);
```

Comment:
Earlier in this file you consistently use `ownerDocument`
Should this be `view.dom.ownerDocument.body.append(...)` instead of the global `document`?